### PR TITLE
Feature core 1.2.0

### PR DESCRIPTION
**Pull request by EdAtkin**
# Pull request description
Getting develop up to date with changes that occurred for v1.2.0. This is mainly changes to the remove the nutype and oscnutype from the samplePDFFDBase object and replacing with an event-by-event level vector of pdg codes for before and after oscillation (longer term this will eliminate the need to split the MC files by oscillation channel). There is an additional handy function for getting the spline interpolation type from the DetID which is handy for the spline loading for the FD. Also there is a bug-fix where DetID was not being initialised correctly. Small fix for reading in the spline binning as well for different delimiters.

## Changes or fixes
- remove of oscnutype and nutype
- vector for neutrino pdg codes
- bug-fix to getting DetID in samplePDFFDBase

### CODE REVIEW

File: covariance/covarianceXsec.cpp [hunk 1]
```diff
@@ -121,6 +122,19 @@ const std::vector<std::string> covarianceXsec::GetSplineParsNamesFromDetID(const
...
+    auto &SplineIndex = pair.first;
+    auto &SystIndex = pair.second;
+    if ((GetParDetID(SystIndex) & DetID )){
+      returnVec.push_back(SplineParams.at(SplineIndex)._SplineInterpolationType);
+    }
+  }
+
+  return returnVec;
+}
+
```
Comment (KSkwarczynski) [inline covariance/covarianceXsec.cpp:137]: Can you try using AppliesToDetID?
https://github.com/mach3-software/MaCh3/blob/1bf48ff1bcaec1c9c15cb9d30057a9312f9dc3ad/covariance/covarianceXsec.cpp#L223C9-L223C23
We gonna have to change det id treamtent and usng function will make process easier

File: samplePDF/samplePDFFDBase.cpp [hunk 2]
```diff
@@ -1381,46 +1384,46 @@ void samplePDFFDBase::SetupNuOscillator() {
...
+        }
+
+        if (std::abs((*MCSamples[iSample].nupdg[iEvent])) == NuPDG::kNue) {
+          FinalFlav = NuOscillator::kElectron;
+        } else if (std::abs((*MCSamples[iSample].nupdg[iEvent])) == NuPDG::kNumu) {
+          FinalFlav = NuOscillator::kMuon;
+        } else if (std::abs((*MCSamples[iSample].nupdg[iEvent])) == NuPDG::kNutau) {
+          FinalFlav = NuOscillator::kTau;
+        }
+
```
Comment (KSkwarczynski) [inline samplePDF/samplePDFFDBase.cpp:1405]: I would simplfy this into
```
InitFlav = Func((*MCSamples[iSample].nupdgUnosc[iEvent])))
FinalFlav  = Func(*MCSamples[iSample].nupdg[iEvent])))
```

Comment (KSkwarczynski) [reply samplePDF/samplePDFFDBase.cpp:1405]: I am nitpicky becasue once we start support sterile this will be terrific

Comment (EdAtkin) [reply samplePDF/samplePDFFDBase.cpp:1405]: yeah that's fair, I'll change this. Don't want to have 3+N if statements!

File: covariance/covarianceXsec.cpp (hunk 1 above)
Comment (EdAtkin) [reply covariance/covarianceXsec.cpp:137]: yep I'll give it a try

**Review by KSkwarczynski — APPROVED** [review]
Merge time

**Pull request merged**